src/lxc/lxccontainer.c

```diff
@@ -2630,16 +2630,40 @@ static bool do_lxcapi_save_config(struct lxc_container *c, const char *alt_file)
 	if (lret)
 		return log_error(false, "Failed to acquire lock");
 
-	fd_config = open(alt_file, O_WRONLY | O_CREAT | O_TRUNC | O_CLOEXEC,
-			 S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP);
-	if (fd_config < 0) {
-		SYSERROR("Failed to open config file \"%s\"", alt_file);
+	char tmpfile[PATH_MAX];
```

Comment:
We don't really know where/how lxc was compiled, so PATH_MAX could be really short.  Worth adding an assert that it will be big enough for the tempfile path.